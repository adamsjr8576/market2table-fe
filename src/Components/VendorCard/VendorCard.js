import React from 'react';
import './VendorCard.scss';
import { images } from "../../images/images"
import { withRouter } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { addSelectedVendor } from '../../actions';



export const VendorCard = ({ id, name, description, image, history }) => {
  const selectedMarketId = useSelector(state => state.selectedMarket);
  const vendors = useSelector(state => state.vendors);
  const vendorInfo = vendors.find(vendor => vendor.id === id);

  const dispatch = useDispatch()
  

  

  const handleVendorPage = () => {
    history.push(`/markets/${selectedMarketId}/vendors/${id}`)
    dispatch(addSelectedVendor(vendorInfo))
  }

  return (
    <article onClick={handleVendorPage}className='vendor-card'>
      <h2 className='vendor-name'>{name}</h2>
      <h2 className='vendor-description'>{description}</h2>
      <img className='vendor-img'src={image}></img>
    </article>
  )
}

export default withRouter(VendorCard);
