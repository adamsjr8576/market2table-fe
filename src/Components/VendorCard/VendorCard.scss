.vendor-card {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  margin-bottom: 35px;
  letter-spacing: 1px;
  height: 420px;
  width: 400px;
  border: 3px solid khaki;
  border-radius: 10px;
  background-color: rgba(0,0,0,0.3);
  padding-left: 10px;
  padding-right: 10px;
  color: cornsilk;
  cursor: pointer;
}

.vendor-card:hover,
.vendor-card:focus {
  transform: scale(1.05);
}
.vendor-img {
    height: 300px;
    width: 350px; 
    border-radius: 10px;
}

.vendor-name {
  margin: 0;
}

.vendor-description{
  font-size: 15px;
  margin: 0;
}