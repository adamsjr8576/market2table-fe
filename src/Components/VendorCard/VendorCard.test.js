import React from 'react';
import { shallow } from 'enzyme';
import VendorCard from './VendorCard';
import { useSelector, useDispatch } from 'react-redux';

const mockState = {
  zipCode: '80401',
  selectedMarket: '1923782',
  vendors: [{name: 'Friendly Famrs', id:'1231', image_link: 'image.png'}]
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState,
  useDispatch: () => jest.fn()
}));

describe('Header', () => {

  let wrapper;
  let mockAddSelectedVendor = jest.fn();
  let mockId = '1231';
  let mockName = 'Friendly Farms';
  let mockDescription = 'A very friendly farm';
  let mockImage = 'friendlyfarm.png';
  let mockHistory = {push: jest.fn()};

  it('should match the snapshot', () => {
    wrapper = shallow(<VendorCard
        addSelectedVendor={mockAddSelectedVendor}
        id={mockId}
        name={mockName}
        description={mockDescription}
        image={mockImage}
        history={mockHistory}
      />);

    expect(wrapper).toMatchSnapshot();
  });
});
