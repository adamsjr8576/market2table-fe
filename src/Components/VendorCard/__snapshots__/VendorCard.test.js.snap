// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorCard should match the snapshot 1`] = `
<article
  className="vendor-card"
  onClick={[Function]}
>
  <h2
    className="vendor-name"
  >
    Friendly Farms
  </h2>
  <h2
    className="vendor-description"
  >
    A very friendly farm
  </h2>
  <img
    className="vendor-img"
    src="friendlyfarm.png"
  />
</article>
`;
